### PR TITLE
[sample_encode] Change membuf functionality

### PR DESCRIPTION
**Pull request by AntonGrishin**
1) Option renamed to perf_opt
2) Preload buffer use until -n frames run out or if -n not specified,
until buffer ends.

### CODE REVIEW

File: samples/sample_encode/include/pipeline_encode.h [hunk 1]
```diff
@@ -161,7 +161,7 @@ struct sInputParams
     bool bSoftRobustFlag;
 
     mfxU32 nTimeout;
-    mfxU16 nMemBuf;
+    mfxU16 nPerfOpt;
```
Comment (dmitryermilov) [inline samples/sample_encode/include/pipeline_encode.h]: please add a comment of the meaning of this variable

File: samples/sample_encode/src/sample_encode.cpp [hunk 2]
```diff
@@ -160,7 +160,7 @@ void PrintHelp(msdk_char *strAppName, const msdk_char *strErrorMessage, ...)
     msdk_printf(MSDK_STRING("   [-WeightedPred:default|implicit ]   - enables weighted prediction mode\n"));
     msdk_printf(MSDK_STRING("   [-WeightedBiPred:default|implicit ] - enables weighted bi-prediction mode\n"));
     msdk_printf(MSDK_STRING("   [-timeout]               - encoding in cycle not less than specific time in seconds\n"));
-    msdk_printf(MSDK_STRING("   [-membuf]                - size of memory buffer in frames\n"));
+    msdk_printf(MSDK_STRING("   [-perf_opt n]              n: number of prefetech frames. In performance mode app preallocates buffer and load first n frames\n"));
```
Comment (mgonchar) [inline samples/sample_encode/src/sample_encode.cpp]: please follow surrounding style. Use something like
```
- sets number of prefetched frames. In performance mode app preallocates buffer and load first n frames\n
```

File: samples/sample_encode/include/pipeline_encode.h [hunk 3]
```diff
@@ -320,7 +320,7 @@ class CEncodingPipeline
     MFXFrameAllocator* m_pMFXAllocator;
     mfxAllocatorParams* m_pmfxAllocatorParams;
     MemType m_memType;
-    mfxU16 m_nMemBuffer;
+    mfxU16 m_nPerfOpt;
```
Comment (dmitryermilov) [inline samples/sample_encode/include/pipeline_encode.h]: please add a comment of the meaning of this variable

File: samples/sample_encode/src/sample_encode.cpp (hunk 2 above)
Comment (AntonGrishin) [reply samples/sample_encode/src/sample_encode.cpp]: Done

File: samples/sample_encode/include/pipeline_encode.h (hunk 3 above)
Comment (AntonGrishin) [reply samples/sample_encode/include/pipeline_encode.h]: Done

File: samples/sample_encode/src/sample_encode.cpp [hunk 4]
```diff
@@ -160,7 +160,7 @@ void PrintHelp(msdk_char *strAppName, const msdk_char *strErrorMessage, ...)
     msdk_printf(MSDK_STRING("   [-WeightedPred:default|implicit ]   - enables weighted prediction mode\n"));
     msdk_printf(MSDK_STRING("   [-WeightedBiPred:default|implicit ] - enables weighted bi-prediction mode\n"));
     msdk_printf(MSDK_STRING("   [-timeout]               - encoding in cycle not less than specific time in seconds\n"));
-    msdk_printf(MSDK_STRING("   [-membuf]                - size of memory buffer in frames\n"));
+    msdk_printf(MSDK_STRING("   [-perf_opt n]            - sets number of prefeteched frames. In performance mode app preallocates buffer and load first n frames\n"));
```
Comment (mgonchar) [inline samples/sample_encode/src/sample_encode.cpp]: typo: prefetched

Comment (AntonGrishin) [reply samples/sample_encode/src/sample_encode.cpp]: done

**Pull request merged**